neuralop/training/trainer.py

```diff
@@ -197,16 +216,32 @@ def train(
 
                 loss = 0.0
 
-                if self.overrides_loss:
+                if self.overrides_training_loss:
                     loss += self.callbacks.compute_training_loss(
                         out=out, **sample, amp_autocast=self.amp_autocast
                     )
                 else:
-                    if self.amp_autocast:
-                        with amp.autocast(enabled=True):
-                            loss += training_loss(out, **sample)
+                    if self.monitor_individual_train_losses:
+                        if self.amp_autocast:
+                            with amp.autocast(enabled=True):
+                                loss_outputs = training_loss(out, **sample)
+                        else:
+                            loss_outputs = training_loss(out, **sample)
+
+                        for loss_name, loss_value in loss_outputs.items():
+                            loss += loss_value
+                            train_losses[loss_name] += loss_value
+                            # accumulate training gradients for each subloss over the epoch
+                            if self.compute_train_grads:
```

Comment:
This is a nice feature, but we typically do not want to do that in every train step but just periodically.

Reply:
Theoretically, we could also save one grad computation (i.e., the `.backward`) by aggregating the grads appropriately, but I think it is not necessary if we only compute it periodically.